src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/openapi/SecurityServersController.java

```diff
@@ -0,0 +1,178 @@
+import org.niis.xroad.centralserver.openapi.model.SecurityServerAddress;
+import org.niis.xroad.centralserver.restapi.converter.SecurityServerConverter;
+import org.niis.xroad.centralserver.restapi.service.SecurityServerService;
+import org.niis.xroad.restapi.config.audit.AuditDataHelper;
+import org.niis.xroad.restapi.config.audit.AuditEventMethod;
+import org.niis.xroad.restapi.config.audit.RestApiAuditEvent;
+import org.niis.xroad.restapi.openapi.BadRequestException;
+import org.niis.xroad.restapi.openapi.ControllerUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.niis.xroad.centralserver.restapi.entity.SecurityServer_.SERVER_CODE;
+
+@RestController
+@RequestMapping(ControllerUtil.API_V1_PREFIX)
+@PreAuthorize("denyAll")
+@RequiredArgsConstructor
+public class SecurityServersController implements SecurityServersApi {
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(SecurityServersController.class);
+
+    private final AuditDataHelper auditData;
+    private final SecurityServerService securityServerService;
+
+    SecurityServerConverter serverConverter = new SecurityServerConverter();
+
+    @Override
+    @PreAuthorize("hasAuthority('DELETE_SECURITY_SERVER')")
+    @AuditEventMethod(event = RestApiAuditEvent.DELETE_SECURITY_SERVER)
+    public ResponseEntity<Void> deleteSecurityServer(String id) {
+        throw new NotImplementedException("deleteSecurityServer not implemented yet");
+    }
+
+    @Override
+    public ResponseEntity<Void> deleteSecurityServerAuthCert(String id, String hash) {
+        throw new NotImplementedException("deleteSecurityServerAuthCert not implemented yet");
+    }
+
+    @Override
+    @Validated
+    @PreAuthorize("hasAuthority('VIEW_SECURITY_SERVERS')")
+    public ResponseEntity<PagedSecurityServers> findSecurityServers(String q, PagingSortingParameters pagingSorting) {
+
+        pagingSorting = getPagingSortingParametersOrDefault(pagingSorting);
+        Sort sorting = getSortOrDefault(pagingSorting);
+
+        Pageable pageable = PageRequest.of(
+                pagingSorting.getOffset(),
+                pagingSorting.getLimit(),
+                sorting);
+
+        var servers = securityServerService.findSecurityServers(q, pageable);
+        return ResponseEntity.ok(serverConverter.convert(servers));
+    }
+
+    private Sort getSortOrDefault(PagingSortingParameters pagingSorting) {
```

Comment:
Could use PageRequestConverter from XRDDEV-1910 instead, it was created as a generic utility for this kind of conversions

Reply:
Addressed in https://github.com/nordic-institute/X-Road/commit/af99d4d47bca9b426372304ff4e5c69309d28b37